pkg/workflow/providers/multicluster/deploy.go

```diff
@@ -82,6 +84,10 @@ func (executor *deployWorkflowStepExecutor) Deploy(ctx context.Context, policyNa
 	if err != nil {
 		return false, "", err
 	}
+	components, err = pkgpolicy.ReplicateComponents(policies, components)
```

Comment:
what will happen if the replicaComponent don't use the replicaKey in the definition template?

Reply:
- If the definition template relies on replicaKey field, an error will be thrown for the incorrect use of that component/trait.
- If the definition template use the replicaKey field and has a fallback logic for handling the lack of replicaKey, it will be fine to use the definition with or without replicaKey.
- If the definition template does not use the replicaKey, nothing will be happened while using the replicaKey. (The component will be repeated dispatched but there is no side effect.)

Reply:
Then it's depends on definitions. In definition it can check if `context.replicaKey` is nil. Just like I did in `replica-webservice.yaml`. But it's not recommended to mix component designed for replication and common ones.

Reply:
what if one component use the replica key in the definition, but it's used in a fixed workload name? It will apply many times with the last config? What about next time? Does it cause any infinitely loop?

Reply:
It is a problem. We can optimize it later. Like scanning the definition used when replicate.

Reply:
Yes, it will apply many times but will not cause infinite loop. Replication does not mutate configurations. Multiple apply will be the same.